### DISCUSSION TITLE
Cost so far today

### DISCUSSION THREAD

**Discussion by nickorr** [amberelectric/public-api#225] [Ideas]
I'm wondering if there would be any way to get the cost value for usage so far today?  I'd be curious have access in Home Assistant to a $ amount which was how much today has cost me so far.

This in the app, but not in the API that I can see.  Presumably would be part of the site API, but maybe would need to be something separate a you'd query it quite regularly - maybe every 5 minutes or so?

I was thinking there's probably a way to estimate this in HA by doing price*usage/60 every minute and adding it up, but the API might be more accurate?

**Comment by madpilot** [comment]
The energy dashboard does a pretty good job of this.

<img width="941" alt="image" src="https://github.com/user-attachments/assets/4b5c88de-2c48-4e60-9e0d-b7dd05bb1678">

You price sensors are already compatible, just setup the general and feed in price

<img width="426" alt="image" src="https://github.com/user-attachments/assets/a65193a1-2e8c-489e-a4d3-b7b85f78934b">

<img width="429" alt="image" src="https://github.com/user-attachments/assets/b815c8c0-1299-451c-9dad-10d7350e6770">

**Reply by rob0101** [reply]
Doesn't the energy dashboard work from 5m estimated prices ?
That makes it very untrustworthy.

![IMG_DAC8C90DD631-1](https://github.com/user-attachments/assets/5bc29eec-beb3-4690-8001-1e16a8923a83)
![Screenshot 2024-10-17 at 11 48 47 am](https://github.com/user-attachments/assets/02c7deef-3472-4994-854c-c450ace0bb56)

**Reply by nickorr** [reply]
I assumed it was something like an interval that summed over time.  I was going to attempt to build one with 1 minute intervals, but even still, if the data is in the app, and could be added to the API, that would be handy.

If it doesn't exist at all or is a long time away, I'll build one and see what it comes up with.

**Reply by rob0101** [reply]
We had a price spike in SA this morning, 20m into a half hour.

The dashboard is much closer to the actual 5m pricing, which is good, but doesn't take into account the massive impact of 30m price averaging when customers pay a higher price than the market is billing.

**Reply by madpilot** [reply]
For Smartshift customers we can do that (we do it already). We can't do it for non-Smartshift customers though, as we don't have the data until the next day.

**Reply by nickorr** [reply]
SmartShift is the battery thing. right?  I don't have a battery, so not sure.  But also noticed in the app, the amount of the cost so far today is not today, it's yesterday's amount.  So it's not there currently for me anyway.

That's all good, I'll see if I can figure out an estimate by doing single minute updates and then see if I can compare it to the actual at the end of the day.

From my perspective then, feel free to close this request off, but I'll leave it open in case it's something that the smartshift people would still like.

Thanks,
Nick

**Reply by madpilot** [reply]
Yes, if you don't have a battery or solar inverter we can talk to we don't get usage data until the next day, so it's just not possible to do this in real-time. (and the values that get dumped out from inverters aren't always accurate).

Going to close this off for now. Thanks!

**Comment by nickorr** [comment]
Thanks, it does show that, but getting that data out of the energy dashboard isn't simple - the only suggestion I could find was to run an SQL query.

So for now at least, I think the feature request is still valuable, and I'd trust the amber data if it was available over the estimated one from HA itself.
